lib/evmone_precompiles/bls.cpp

```diff
@@ -12,57 +12,65 @@ namespace
 /// https://eips.ethereum.org/EIPS/eip-2537#field-elements-encoding
 constexpr auto FP_BYTES_OFFSET = 64 - 48;
 
+/// Validates that integer encoded in big endian is valid element of BLS12-381 fp field
+[[nodiscard]] std::optional<blst_fp> validate_fp_field_element(const uint8_t _p[64])
+{
+    if (intx::be::unsafe::load<intx::uint512>(_p) < BLS_FIELD_MODULUS)
```

Comment:
Reverse condition with quick return. 
```suggestion
    if (intx::be::unsafe::load<intx::uint512>(_p) >= BLS_FIELD_MODULUS)
        return std::nullopt;
```